File: c/b10773.c
#include <stdio.h>
#define STACK_MAX 100000

// 10773: 제로

static int stack[STACK_MAX] = {0,};
static int stack_size = -1;

int isfull(void) {
	if (stack_size == STACK_MAX) {
		return 1;
	}
	return 0;
}

int isempty(void) {
	if (stack_size == -1) {
		return 1;
	}
	return 0;
}

void push(int n) {
	if (isfull() == 1) {
		return;
	}
	stack_size += 1;
	stack[stack_size] = n;
}

int pop() {
	int n;
	if (isempty() == 1) {
		return 0;
	}
	n = stack[stack_size];
	stack_size -= 1;
	return n;
}

int main(void) {
	int K, i, n;
	int result = 0;

	scanf("%d", &K);
	for (i = 0; i < K; i++) {
		scanf("%d", &n);
		if (n == 0) {
			pop();
		} else {
			push(n);
		}
	}

	n = stack_size + 1;
	for (i = 0; i < n; i++) {
		result += pop();
	}

	printf("%d", result);
	return 0;
}

File: python/b10773.py

# 10773: 제로

stack = list()
stack_size = -1
stack_max = 100000

def isfull():
	global stack_size
	global stack_max
	if stack_size == stack_max:
		return True
	return False

def isempty():
	global stack_size
	if stack_size == -1:
		return True
	return False

def push(n):
	global stack_size
	if isfull():
		return
	stack_size += 1
	stack.append(n)

def pop():
	global stack_size
	if isempty():
		return -1
	stack_size -= 1
	return stack.pop()

K = int(input())

for i in range(K):
	n = int(input())
	if n == 0:
		pop()
	else:
		push(n)

result = 0
for i in range(stack_size+1):
	result += pop()
print(result)